File: 🔴 Hard/995 - Minimum Number of K Consecutive Bit Flips/_995MinNoOfKConsBitFlips.js
var minKBitFlips = function (nums, k) {
    let n = nums.length;
    let w = 0;
    let count = 0;
    let flipped = [];

    for (let i = 0; i < n; i++) {
        if (i >= k) {
            if (flipped[i - k]) {
                w--;
            }
        }
        if (w % 2 == nums[i]) {
            if (i + k > n) {
                return -1;
            }

            w++;
            flipped[i] = true;
            count++;
        }
    }

    return count;
};

File: 🔴 Hard/995 - Minimum Number of K Consecutive Bit Flips/_995MinNoOfKConsBitFlips.c
int minKBitFlips(int* nums, int numsSize, int k) {
    int w = 0;
    int count = 0;
    bool flipped[numsSize];

    for (int i = 0; i < numsSize; i++) {
        flipped[i] = false;
    }

    for (int i = 0; i < numsSize; i++) {
        if (i >= k) {
            if (flipped[i - k]) {
                w--;
            }
        }
        if (w % 2 == nums[i]) {
            if (i + k > numsSize) {
                return -1;
            }

            w++;
            flipped[i] = true;
            count++;
        }
    }

    return count;
}